### PR TITLE
Add a go module

### PR DESCRIPTION
**Pull request by ThatCopy**
I couldn't make a optional parameter and you can't make a `const `array in go, but 
> An array isn't immutable by nature; you can't make it constant.

### CODE REVIEW

**Comment by ThatCopy** [comment]
I had to use CamelCase, because go its very strict of x-case. I can use kebab-case but go's linting doesn't like that.

File: go/dousername.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	"strings"
+	"time"
+)
+
+func random(arr []string) string {
+	rand.Seed(time.Now().UnixNano())
+	return arr[rand.Intn(len(arr))]
+}
+
+func randomDesc(seaCreaturess []string, randNoun string, descriptors []string, verbs []string) string {
```
Comment (MattIPv4) [inline go/dousername.go]: Why do we pass in `seaCreaturess`, `descriptors` and `verbs` here? Can we just not use the arrays directly?

File: go/dousername.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+// generate : Returns a random DO username
+func generate() string {
+	randNoun := random(SeaList)
+	randDesc := randomDesc(SeaCreatures, randNoun, Descriptors, CreatuteDescriptors)
+	randColor := random(Colors)
+
+	if len(randDesc+randNoun+randColor) <= 30 {
+		return capitalize([]string{randDesc, randColor, randNoun})
+	} else if len(randDesc+randColor) <= 30 {
```
Comment (MattIPv4) [inline go/dousername.go]:
```suggestion
	} else if len(randDesc+randNoun) <= 30 {
```

File: go/dousername.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		return capitalize([]string{randDesc, randColor})
```
Comment (MattIPv4) [inline go/dousername.go]:
```suggestion
		return capitalize([]string{randDesc, randNoun})
```

File: go/dousername.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	} else if len(randColor+randNoun) <= 30 {
+		return capitalize([]string{randColor, randColor})
```
Comment (MattIPv4) [inline go/dousername.go]:
```suggestion
		return capitalize([]string{randColor, randNoun})
```

File: go/dousername.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	} else {
+		return capitalize([]string{randNoun})
```
Comment (MattIPv4) [inline go/dousername.go]: This needs to be sliced to `30`

File: go/go.mod [hunk 6]
```diff
@@ -0,0 +1,3 @@
+module github.com/ThatCopy/go-do-namegen
```
Comment (MattIPv4) [inline go/go.mod]: This needs updating to point to this repo

File: go/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+## Example
+
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/ThatCopy/go-do-namegen"
```
Comment (MattIPv4) [inline go/README.md]: This needs updating to point to this repo

File: go/README.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+func main() {
+	fmt.Println(dousername.GetName())
```
Comment (MattIPv4) [inline go/README.md]:
```suggestion
	fmt.Println(dousername.Generate())
```

File: go/dousername.go [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+var Colors = []string{"blue", "blueGreen", "darkCyan", "electricBlue", "greenBlue", "lightCyan", "lightSeaGreen", "seaGreen", "turquoise", "aqua", "aquamarine", "teal", "cyan", "gray", "darkBlue", "cerulean", "azure", "lapis", "navy"}
+
+// SeaList : a combination of the creature + object nouns
+var SeaList = append(SeaCreatures, SeaObjects...)
+
+// Descriptors : combination of the generic (adjective + size) descriptors
+var Descriptors = append(AdjectiveDescriptors, SizeDescriptors...)
+
+// generate : Returns a random DO username
+func generate() string {
```
Comment (MattIPv4) [inline go/dousername.go]:
```suggestion
func Generate() string {
```
Needs to be PascalCase to be public iirc

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

**Review by MattIPv4 — APPROVED** [review]
Sorry for not getting to this sooner, not sure why I lost track of updates here a while back. LGTM, tyvm!

**Pull request merged**